docs/flows/flows.puml

```diff
@@ -298,9 +413,124 @@ bitdaoadmin <-- bitdaogovernance: pass/fail vote result
 
 bitdaoadmin -> bitdaogovernance: start execution
 executor <-- bitdaogovernance: execute proposal
-executor -> governanceproxy: set new bitdaogovernance implementation
-executor -> treasuryproxy: set new bitdaotreasury implementation
-executor -> executorproxy: set new executor implementation
+executor -> proxyadmin: call token proxy upgrade to new token implementation
+
+newpage
+
+autonumber "UpgradeSnapshot:[00]"
+
+== Upgrade Snapshot ==
+
+alice -> forum: create snapshot upgrade request forum proposal
+activate forum
+
+guarantor -> forum: get proposals
+guarantor <-- forum: list of proposals
+guarantor -> forum: put positive/negative feedback
+
+community -> forum: get proposals
+community <-- forum: list of proposals
+community -> forum: put positive/negative feedback
+
+alice <-- forum: get positive/negative feedback
+deactivate forum
+
+community --> community: delegate votes
+
+alice -> snapshot: submit snapshot upgrade request proposal
+activate snapshot
+guarantor -> snapshot: get proposals
+guarantor <-- snapshot: list of proposals
+guarantor -> snapshot: vote to release or proportional slash
+
+tokenholders -> snapshot: get proposals
+tokenholders <-- snapshot: list of proposals
+tokenholders -> snapshot: vote on open proposals
+tokenholders <-- snapshot: vote confirmation
+deactivate snapshot
+
+bitdaoadmin -> snapshot: get vote result
+bitdaoadmin <-- snapshot: pass/fail vote result
+bitdaoadmin -> snapshot: upgrade snapshot
```

Comment:
thought: I think this flow works great. My (non-blocking) question is whether we should (or can express) an ops flow for upgrading off-chain systems since the admins and participants there are like traditional web2 flows. wdyt?

Reply:
Still I think its okay to have that as we have the same for other flows where we do off-chain actions with snapshot and forum, then it makes sense to indicate it for the upgrade flow as well. It allows to track all flows even for off-chain and update them and possibility to rely on them, when we will evolve them or move on-chain.